proxy-node-shared/src/main/java/uk/gov/ida/notification/shared/proxy/MetatronProxy.java

```diff
@@ -0,0 +1,39 @@
+package uk.gov.ida.notification.shared.proxy;
+
+import uk.gov.ida.exceptions.ApplicationException;
+import uk.gov.ida.notification.contracts.CountryMetadataResponse;
+import uk.gov.ida.notification.exceptions.proxy.MetatronResponseException;
+
+import javax.ws.rs.core.UriBuilder;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+
+public class MetatronProxy {
+    private final ProxyNodeJsonClient metatronClient;
+    private final URI metatronUri;
+
+    public MetatronProxy(
+            ProxyNodeJsonClient metatronClient,
+            URI metatronUri) {
+        this.metatronClient = metatronClient;
+        this.metatronUri = metatronUri;
+    }
+
+    public CountryMetadataResponse getCountryMetadata(String entityId) {
```

Comment:
It bugs me that entityId is a String when it could possibly be a URI, but its used as a String is throughout our codebase. Its a me problem 😢

Reply:
Yeah, I keep thinking the same thing! I think it boils down to that we don't actually use the entityId as URL, just as a label or identifier, so we don't need all the bells and whistles that come with having a URI.